compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -287,7 +287,9 @@ pub fn intern_const_alloc_for_constprop<
         return Ok(());
     }
     // Move allocation to `tcx`.
-    for _ in intern_shallow(ecx, alloc_id, Mutability::Not).map_err(|()| err_ub!(DeadLocal))? {
+    if let Some(_) =
+        (intern_shallow(ecx, alloc_id, Mutability::Not).map_err(|()| err_ub!(DeadLocal))?).next()
+    {
```

Comment:
This is wrong. We don't check all elements anymore, only the first one

Reply:
> We don't check all elements anymore, only the first one

That's what we do here.

Reply:
Yea, what I mean is that before this PR we checked all elements

Reply:
Isn't [this panic](https://github.com/onur-ozkan/rust/blob/774ae599aba35f7b323e77be88923b8f476ee783/compiler/rustc_const_eval/src/interpret/intern.rs#L294-L298) stop the iteration at the first element?

Reply:
Oh.. I somehow hallucinated a different behaviour.